### PR TITLE
Feature/pxweb2 223 refactor app.tsx

### PR DESCRIPTION
**Pull request by PerIngeVaaje**

### CODE REVIEW

File: apps/pxweb2/src/app/components/Presentation/Presentation.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        !isLoadingMetadata &&
+        !initialRun
+      ) {
+        tableData.fetchTableData(tableId ? tableId : 'tab1292', i18n);
+        setIsMissingMandatoryVariables(false);
+      }
+      if (!hasSelectedMandatoryVariables && !initialRun) {
+        setIsMissingMandatoryVariables(true);
+      }
+      if (initialRun) {
```
Comment (MikaelNordberg) [inline apps/pxweb2/src/app/components/Presentation/Presentation.tsx:59]: Should this be here?

File: apps/pxweb2/src/app/components/Presentation/Presentation.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    // const SelectedValues = variables.getNumberOfSelectedValues();
+
+    const hasSelectedMandatoryVariables = pxTableMetadata?.variables
+      .filter((variable) => variable.mandatory)
+      .every((variable) =>
+        selectedVBValues.some(
+          (selectedVariable) => selectedVariable.id === variable.id
+        )
+      );
+    if (initialRun && !hasSelectedValues) {
```
Comment (MikaelNordberg) [inline apps/pxweb2/src/app/components/Presentation/Presentation.tsx:43]: Maybe add comments about that we will replace this with dataloader?

File: apps/pxweb2/src/app/components/Selection/Selection.module.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+@use '$ui/style-dictionary/dist/scss/fixed-variables.scss' as fixed;
+.variableListContainer {
```
Comment (MikaelNordberg) [inline apps/pxweb2/src/app/components/Selection/Selection.module.scss:2]: I think this one is removed. Styling is done directly on VariableList i think

File: apps/pxweb2/src/app/components/Selection/Selection.module.scss [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    display: flex;
+    flex-direction: column;
+    gap: fixed.$spacing-2;
+  }
+  .scrollable {
```
Comment (MikaelNordberg) [inline apps/pxweb2/src/app/components/Selection/Selection.module.scss:7]: Scrollable has been removed

File: apps/pxweb2/src/main.tsx [hunk 5]
```diff
@@ -33,13 +33,13 @@ if (location.href.indexOf('localhost') > -1) {
 }
 
 root.render(
-  <StrictMode>
+  // <StrictMode>
```
Comment (MikaelNordberg) [inline apps/pxweb2/src/main.tsx:36]: Should not be removed

File: apps/pxweb2/src/mappers/TableSelectionResponseMapper.ts [hunk 6]
```diff
@@ -5,6 +5,7 @@ export function mapTableSelectionResponse(
   response: SelectionResponse
 ): SelectedVBValues[] {
   // TODO:  Check if selection.selection is intended,
+  // console.log('response i MAPPER=' + response.selection.selection[0].variableCode)
```
Comment (MikaelNordberg) [inline apps/pxweb2/src/mappers/TableSelectionResponseMapper.ts:8]: Remove logging

**Review by MikaelNordberg — CHANGES_REQUESTED** [review]
Look at my comments and see you need to make some changes

**Comment by MikaelNordberg** [comment]
I tested and saw that variable- and value names got the wrong texts when changing language

File: apps/pxweb2/src/app/app.tsx [hunk 7]
```diff
@@ -516,47 +52,18 @@ export function App() {
...
-              {selectedNavigationView === 'edit' && drawerEdit}
-              {selectedNavigationView === 'save' && drawerSave}
-              {selectedNavigationView === 'help' && drawerHelp}
-            </NavigationDrawer>
-          )}
+          <Selection
+            selectedNavigationView={selectedNavigationView}
+            selectedTabId={selectedTableId}
+            setSelectedNavigationView={changeSelectedNavView}
+            setSelectedTableId={setSelectedTableId}
```
[the remark this answers is not in the document]
Comment (PerIngeVaaje) [reply apps/pxweb2/src/app/app.tsx]: Removed  setSelectedTableId

File: apps/pxweb2/src/app/app.tsx [hunk 8]
```diff
@@ -516,47 +52,18 @@ export function App() {
...
-                      descriptionTxt={t(
-                        'presentation_page.main_content.table.warnings.missing_mandatory.description'
-                      )}
-                    />
-                  )}
-                </>
-              )}{' '}
+              <Presentation
+                selectedTabId={selectedTableId}
+                selectedNavigationView={selectedNavigationView}
```
[the remark this answers is not in the document]
Comment (PerIngeVaaje) [reply apps/pxweb2/src/app/app.tsx]: Removed selectedNavigationView

File: apps/pxweb2/src/app/components/Presentation/Presentation.tsx (hunk 1 above)
Comment (PerIngeVaaje) [reply apps/pxweb2/src/app/components/Presentation/Presentation.tsx:56]: I think it should be here

File: apps/pxweb2/src/app/components/Presentation/Presentation.tsx (hunk 2 above)
Comment (PerIngeVaaje) [reply apps/pxweb2/src/app/components/Presentation/Presentation.tsx:40]: Comment added

File: apps/pxweb2/src/app/components/Selection/Selection.module.scss (hunk 3 above)
Comment (PerIngeVaaje) [reply apps/pxweb2/src/app/components/Selection/Selection.module.scss]: Style removed

File: apps/pxweb2/src/app/components/Selection/Selection.module.scss (hunk 4 above)
Comment (PerIngeVaaje) [reply apps/pxweb2/src/app/components/Selection/Selection.module.scss]: Style removed

File: apps/pxweb2/src/main.tsx (hunk 5 above)
Comment (PerIngeVaaje) [reply apps/pxweb2/src/main.tsx]: Removed comment out

File: apps/pxweb2/src/mappers/TableSelectionResponseMapper.ts (hunk 6 above)
Comment (PerIngeVaaje) [reply apps/pxweb2/src/mappers/TableSelectionResponseMapper.ts]: Removed

**Review by michaelpande — APPROVED** [review]
LGTM 👍

**Review by MikaelNordberg — APPROVED** [review]
Nice work! Looks good

**Pull request merged**